legend-engine-core/legend-engine-core-language-pure/legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/data/core/EmbeddedDataCompilerExtension.java

```diff
@@ -122,10 +112,37 @@ else if (embeddedData instanceof DataElementReference
         }
     }
 
-    private static boolean validatePairForModelStoreData(PackageableElementPtr value, CompileContext context)
+    static boolean validatePairForModelStoreData(PackageableElementPtr value, CompileContext context)
     {
         return (context.pureModel.getPackageableElement(value.fullPath) instanceof Root_meta_pure_data_DataElement &&
                 ((Root_meta_pure_data_DataElement) context.pureModel.getPackageableElement(value.fullPath))._data() instanceof Root_meta_external_format_shared_metamodel_data_ExternalFormatData
         );
     }
+    
+    default Iterable<? extends Function2<DataElementReference, PureModelContextData, List<EmbeddedData>>> getExtraDataElementReferencePMCDTraversers()
+    {
+        return Collections.emptyList();
+    }
+    
+    static EmbeddedData getEmbeddedDataFromDataElement(DataElementReference dataElementReference, PureModelContextData pureModelContextData)
+    {
+        List<EmbeddedData> dataList = ListIterate
+                .select(CompilerExtensions.fromAvailableExtensions().getExtensions(), e -> e instanceof EmbeddedDataCompilerExtension)
```

Comment:
You could have done selectInstaceOf and this will do the select and the collect all in one call.